dynamo-circe/src/main/scala/d4s/codecs/circe/D4SCirceDecoder.scala

```diff
@@ -2,48 +2,36 @@ package d4s.codecs.circe
 
 import cats.implicits._
 import d4s.codecs.CodecsUtils.{CannotDecodeAttributeValueAsJson, CirceDecodeException, DynamoDecoderException}
-import d4s.codecs.circe.DynamoDecoder.attributeToJson
+import d4s.codecs.D4SDecoder
 import io.circe.{Decoder, Json}
 import software.amazon.awssdk.core.util.{DefaultSdkAutoConstructList, DefaultSdkAutoConstructMap}
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue
 
 import scala.jdk.CollectionConverters._
 
-class DynamoDecoder[T: Decoder] {
-
-  /** Not typesafe. This will only succeed if `T` is encoded as a JsonObject (has ObjectEncoder instance) */
-  def decode(item: Map[String, AttributeValue]): Either[DynamoDecoderException, T] = {
-    for {
-      json <- item.toList
-        .traverse(_.traverse(attributeToJson)).map(Json.fromFields(_))
-        .toRight(new CannotDecodeAttributeValueAsJson(item.toString))
-      res <- json.as[T].left.map(new CirceDecodeException(item.toString, json, _))
-    } yield res
-  }
-
-  final def decode(item: java.util.Map[String, AttributeValue]): Either[DynamoDecoderException, T] = {
-    decode(item.asScala.toMap)
-  }
-
-  /** This will succeed even if `T` is not encoded as a JsonObject */
-  def decodeAttribute(v: AttributeValue): Either[DynamoDecoderException, T] = {
-    attributeToJson(v)
-      .toRight(new CannotDecodeAttributeValueAsJson(v.toString))
-      .flatMap(json => json.as[T].left.map(new CirceDecodeException(v.toString, json, _)))
+trait D4SCirceDecoder[T] extends D4SDecoder[T]
+
+object D4SCirceDecoder {
+  def apply[A: D4SCirceDecoder]: D4SCirceDecoder[A] = implicitly
+  def derived[T: Decoder]: D4SCirceDecoder[T] = new D4SCirceDecoder[T] {
+    /** Not typesafe. This will only succeed if `T` is encoded as a JsonObject (has ObjectEncoder instance) */
```

Comment:
Hmm, the distinction between `D4SAttributeDecoder` & `D4SDecoder` is meaningless - `decode` is exactly the same as `decodeAttribute(AttributeBuilder.m(value).build())`, I think it may be best to rename `D4SAttributeDecoder`->`D4SDecoder` and remove the redundant trait.